--- texinfo.orig/tp/Texinfo/Convert/HTML.pm
+++ texinfo/tp/Texinfo/Convert/HTML.pm
@@ -9927,6 +9927,8 @@
   if (!$fulltitle and $self->{'global_commands'}->{'titlefont'}
       and $self->{'global_commands'}->{'titlefont'}->[0]->{'args'}
       and defined($self->{'global_commands'}->{'titlefont'}->[0]->{'args'}->[0])
+      and $self->{'global_commands'}->{'titlefont'}->[0]
+                                                ->{'args'}->[0]->{'contents'}
       and @{$self->{'global_commands'}->{'titlefont'}->[0]->{'args'}->[0]->{'contents'}}) {
     $fulltitle = $self->{'global_commands'}->{'titlefont'}->[0];
   }
