 ChangeLog     |    7 +++++++
 util/texi2dvi |    4 +++-
 2 files changed, 10 insertions(+), 1 deletions(-)

--- texinfo-4.12.dfsg.1.orig/util/texi2dvi
+++ texinfo-4.12.dfsg.1/util/texi2dvi
@@ -141,11 +141,13 @@ test -n "$TEX" && test -d "$TEX" && unset TEX
   }
   test_local >/dev/null 2>&1
   test $foo = bar
-) || local () {
+) || eval '
+local () {
   case $1 in
     *=*) eval "$1";;
   esac
 }
+'
 
 
 # cd_orig
